infrastructure/security.tf

```diff
@@ -214,6 +223,85 @@ resource "aws_security_group_rule" "data_refinery_db_outbound" {
   security_group_id = "${aws_security_group.data_refinery_db.id}"
 }
 
+##
+# PGBouncer
+##
+
+resource "aws_security_group" "data_refinery_pg" {
+  name = "data-refinery_pg-${var.user}-${var.stage}"
+  description = "data_refinery_pg-${var.user}-${var.stage}"
+  vpc_id = "${aws_vpc.data_refinery_vpc.id}"
+
+  tags {
+    Name = "data-refinery-pg-${var.user}-${var.stage}"
+  }
+}
+
+resource "aws_security_group_rule" "data_refinery_pg_ssh" {
+  type = "ingress"
+  from_port = 22
+  to_port = 22
+  protocol = "tcp"
+  cidr_blocks = ["0.0.0.0/0"]
+  security_group_id = "${aws_security_group.data_refinery_pg.id}"
+}
+
+
+resource "aws_security_group_rule" "data_refinery_pg_workers_tcp" {
+  type = "ingress"
+  from_port = 0
+  to_port = 65535
+  protocol = "tcp"
+  source_security_group_id = "${aws_security_group.data_refinery_worker.id}"
+  security_group_id = "${aws_security_group.data_refinery_pg.id}"
+}
+
+resource "aws_security_group_rule" "data_refinery_pg_foreman_tcp" {
+  type = "ingress"
+  from_port = 0
+  to_port = 65535
+  protocol = "tcp"
+  source_security_group_id = "${aws_security_group.data_refinery_foreman.id}"
+  security_group_id = "${aws_security_group.data_refinery_pg.id}"
+}
+
+resource "aws_security_group_rule" "data_refinery_pg_api_tcp" {
+  type = "ingress"
+  from_port = 0
+  to_port = 65535
+  protocol = "tcp"
+  source_security_group_id = "${aws_security_group.data_refinery_api.id}"
+  security_group_id = "${aws_security_group.data_refinery_pg.id}"
+}
+
+resource "aws_security_group_rule" "data_refinery_pg_workers_icmp" {
+  type = "ingress"
+  from_port = -1
+  to_port = -1
+  protocol = "icmp"
+  source_security_group_id = "${aws_security_group.data_refinery_worker.id}"
+  security_group_id = "${aws_security_group.data_refinery_pg.id}"
+}
+
+resource "aws_security_group_rule" "data_refinery_pg_api_icmp" {
+  type = "ingress"
+  from_port = -1
+  to_port = -1
+  protocol = "icmp"
+  source_security_group_id = "${aws_security_group.data_refinery_api.id}"
+  security_group_id = "${aws_security_group.data_refinery_pg.id}"
+}
```

Comment:
Two questions about all these security group rules:
* I think we already have these in place for communicating directly with the rds instance. Is there a reason our clients would need access to both the rds instance and the pgbouncer instance?
* The workers and api have both tcp and icmp rules, is there a reason the foreman doesn't need icmp?

Reply:
For debugging, I think it'd be useful to have direct psql access. We mostly just want PGB for Django.